pypdf/_page.py

```diff
@@ -473,9 +532,50 @@ def _get_image(
         if isinstance(id, List) and len(id) == 1:
             id = id[0]
         try:
-            xobjs = cast(
-                DictionaryObject, cast(DictionaryObject, obj[PG.RESOURCES])[RES.XOBJECT]
-            )
+            if isinstance(id, str) and id.find(RES.PATTERN) == 0:
```

Comment:
with my naming proposal you may simplify the following section to make xobjs to point the proper node above the images this would prevent changing the code at new  line 589